src/app_view.js

```diff
@@ -0,0 +1,277 @@
+/**
+ * @license
+ * Blockly Demos: Block Factory
+ *
+ * Copyright 2017 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+goog.require('BlockFactory');
+goog.require('FactoryUtils');
+goog.require('BlockLibraryController');
+goog.require('BlockExporterController');
+goog.require('goog.dom.classlist');
+goog.require('goog.ui.PopupColorPicker');
+goog.require('goog.ui.ColorPicker');
+
+/**
+ * @fileoverview The AppView Class deals with the visual parts of the main
+ * devtools application, such as the menubar.
+ *
+ * @author celinechoo (Celine Choo)
+ */
+class AppView {
+  constructor() {
+    // Main window of application.
+    this.win = nw.Window.get();
+
+    // Initializes menu structure. Leaf nodes are actionable MenuItems.
+    this.menuTree = {
+      'File': {
+        'New': {
+          'New Project': null,
+          'New Block': null,
+          'New Library': null,
+          'New Toolbox': null
+        },
+        'Open Project': null,
+        'Save': null,
+        'Save Only': {
+          'Save as Web Only': null,
+          'Save as iOS Only': null,
+          'Save as Android Only': null
+        },
+        'Import': {
+          'Blocks': null,
+          'Library': null,
+          'Toolbox': null,
+          'Workspace': null
+        },
+        'Create Application': {
+          'All Applications': null,
+          'Blockly Web': null,
+          'Blockly iOS': null,
+          'Blockly Android': null
+        }
+      },
+      'Edit': {},
+      'Help': {}
+    }
+
+    // Initializes menubar.
+    this.setMenu();
+  }
+
+  /**
+   * Initializes menubar for application. Contains File, Edit, Help with
+   * dropdowns to more options such as save, export, etc.
+   */
+  setMenu() {
+    let menu = new nw.Menu({type: 'menubar'});
+    let file = new nw.Menu();
+    let edit = new nw.Menu();
+    let help = new nw.Menu();
+
+    // FILE > NEW dropdown.
+    let file_new = new nw.Menu();
+    this.menuTree['File']['New']['New Project'] =
```

Comment:
The duplication of the menu hierarchy here and above bothers me.  There is an opportunity for the two to get out of sync. In this first one item, you've specified 'New Project' three times: once above, once to the left of the assignment, and again as a parameter.

Consider adjusting your `addMenuAction` helper function to actually use `path` as a paththat will take in a menu path (string array), and find or generate any necessary menuTree hierarchy that doesn't already exist. Something like...

```
function addMenuAction(pathToItem, clickHandler) { ... }

addMenuAction(['File', 'New', 'Project'], () => { ... })
```

Then, there is only one place to edit it each path.